### PR TITLE
refactor: Remove or comment out useless code in the tskv directory.

### PR DESCRIPTION
**Pull request by liikeuforever**
# Required checklist
- [ ] Sample config files updated (`config`,`meta/config` and `default config`) 
- [ ] If there are user-facing changes, the documentation needs to be updated prior to approving the PR( [Link]() )
- [ ] If there are any breaking changes to public APIs, please add the `api change` label.
- [ ] Signed [CLA](https://cla-assistant.io/cnosdb/cnosdb) (if not already signed)

# Which issue does this PR close?

[//]: # (We generally require a GitHub issue to be filed for all bug fixes and enhancements and this helps us generate change logs for our releases. You can link an issue to this PR using the GitHub syntax. For -- example `Closes #123` indicates that this PR will close issue #123.)

Related #.
Remove tskv/lib.rs in the：
```
#![allow(dead_code)]
#![allow(unreachable_patterns)]
```
Clearing warnings that appear in clippy
# Rationale for this change

[//]: # (Why are you proposing this change? If this is already explained clearly in the issue then this section is not needed.
 Explaining clearly why changes are proposed helps reviewers understand your changes and offer better suggestions for fixes.)

# Are there any user-facing changes?

[//]: # (There is no need to duplicate the description in the issue here, but it is sometimes worth providing a summary of the individual changes in this PR.)

### CODE REVIEW

File: tskv/src/compaction/check.rs [hunk 1]
```diff
@@ -14,73 +12,16 @@ use crate::tseries_family::TseriesFamily;
 use crate::TseriesFamilyId;
 
 /// Duration of each TimeRangeHashTreeNode, 24 hour.
-const DEFAULT_DURATION: i64 = 24 * 60 * 60 * 1_000_000_000;
+const _DEFAULT_DURATION: i64 = 24 * 60 * 60 * 1_000_000_000;
```
Comment (Subsegment) [inline tskv/src/compaction/check.rs:15]: rm this

File: tskv/src/compaction/compact.rs [hunk 2]
```diff
@@ -535,13 +526,13 @@ pub(crate) struct CompactIterator {
     tsm_readers: Vec<Arc<TsmReader>>,
     compacting_files: BinaryHeap<Pin<Box<CompactingFile>>>,
     /// Maximum values in generated CompactingBlock
-    max_data_block_size: usize,
+    _max_data_block_size: usize,
```
Comment (Subsegment) [inline tskv/src/compaction/compact.rs:529]: rm this

File: tskv/src/compaction/compact.rs [hunk 3]
```diff
@@ -535,13 +526,13 @@ pub(crate) struct CompactIterator {
...
     // /// The time range of data to be merged of level-0 data blocks.
     // /// The level-0 data that out of the thime range will write back to level-0.
     // level_time_range: TimeRange,
     /// Decode a data block even though it doesn't need to merge with others,
     /// return CompactingBlock::DataBlock rather than CompactingBlock::Raw .
-    decode_non_overlap_blocks: bool,
+    _decode_non_overlap_blocks: bool,
```
Comment (Subsegment) [inline tskv/src/compaction/compact.rs:535]: rm this

File: tskv/src/compaction/check.rs [hunk 4]
```diff
@@ -101,41 +42,14 @@ impl std::fmt::Display for VnodeHashTreeNode {
 pub struct FieldHashTreeNode {
     pub field_id: FieldId,
     pub time_ranges: Vec<TimeRangeHashTreeNode>,
-    min_ts: Timestamp,
-    max_ts: Timestamp,
+    _min_ts: Timestamp,
+    _max_ts: Timestamp,
```
Comment (Subsegment) [inline tskv/src/compaction/check.rs:46]: rm this

File: tskv/src/compaction/check.rs [hunk 5]
```diff
@@ -14,73 +12,16 @@ use crate::tseries_family::TseriesFamily;
...
-            hasher.update(&h);
-        }
-        hasher.finalize().into()
-    }
-
-    pub fn len(&self) -> usize {
-        self.fields.iter().map(|f| f.len()).sum()
-    }
+    _min_ts: Timestamp,
+    _max_ts: Timestamp,
```
Comment (Subsegment) [inline tskv/src/compaction/check.rs:24]: rm this

File: tskv/src/compaction/flush.rs [hunk 6]
```diff
@@ -22,8 +22,8 @@ use crate::{ColumnFileId, TsKvContext, TseriesFamilyId, TskvError};
 pub struct FlushTask {
     ts_family_id: TseriesFamilyId,
     mem_caches: Vec<Arc<RwLock<MemCache>>>,
-    low_seq_no: u64,
-    high_seq_no: u64,
+    _low_seq_no: u64,
+    _high_seq_no: u64,
```
Comment (Subsegment) [inline tskv/src/compaction/flush.rs:26]: rm this

File: tskv/src/compaction/picker.rs [hunk 7]
```diff
@@ -17,8 +17,8 @@ pub trait Picker: Send + Sync + Debug {
 /// Compaction picker for picking files in level
 #[derive(Debug)]
 pub struct LevelCompactionPicker {
-    picking: AtomicBool,
-    storage: Arc<StorageOptions>,
+    _picking: AtomicBool,
+    _storage: Arc<StorageOptions>,
```
Comment (Subsegment) [inline tskv/src/compaction/picker.rs:21]: How about rm this? @zipper-meng

File: tskv/src/file_system/file/async_file.rs [hunk 8]
```diff
@@ -118,7 +118,7 @@ impl FsRuntime {
 
 pub struct AsyncFile {
     inner: RawFile,
-    ctx: Arc<FsRuntime>,
+    _ctx: Arc<FsRuntime>,
```
Comment (Subsegment) [inline tskv/src/file_system/file/async_file.rs:121]: How about rm this? @roseboy-liu

File: tskv/src/index/engine.rs [hunk 9]
```diff
@@ -11,7 +11,7 @@ use super::{IndexError, IndexResult};
 
 #[derive(Debug)]
 pub struct IndexEngine {
-    dir: PathBuf,
+    _dir: PathBuf,
```
Comment (Subsegment) [inline tskv/src/index/engine.rs:14]: rm this

File: tskv/src/index/binlog.rs [hunk 10]
```diff
@@ -300,7 +300,7 @@ impl BinlogWriter {
 }
 
 pub struct BinlogReader {
-    id: u64,
+    _id: u64,
```
Comment (Subsegment) [inline tskv/src/index/binlog.rs:303]: rm this

File: tskv/src/reader/column_group/mod.rs [hunk 11]
```diff
@@ -75,7 +75,7 @@ impl ColumnGroupReader {
         })
     }
 
-    pub fn new_with_unchecked(
+    pub fn _new_with_unchecked(
```
Comment (Subsegment) [inline tskv/src/reader/column_group/mod.rs:78]: rm this

File: tskv/src/reader/iterator.rs [hunk 12]
```diff
@@ -952,53 +871,25 @@ impl QueryOption {
...
     /// Super version of vnode_id, maybe None.
     super_version: Option<Arc<SuperVersion>>,
     /// List of series id filtered from engine.
     series_ids: Arc<Vec<SeriesId>>,
-    series_iter_receiver: Receiver<Option<TskvResult<RecordBatch>>>,
+    _series_iter_receiver: Receiver<Option<TskvResult<RecordBatch>>>,
     series_iter_closer: CancellationToken,
     /// Whether this iterator was finsihed.
-    is_finished: bool,
+    _is_finished: bool,
```
Comment (Subsegment) [inline tskv/src/reader/iterator.rs:886]: rm this

File: tskv/src/reader/page.rs [hunk 13]
```diff
@@ -30,7 +30,7 @@ pub trait PageReader {
 
 #[derive(Clone)]
 pub struct PrimitiveArrayReader {
-    date_type: PhysicalDType,
+    _date_type: PhysicalDType,
```
Comment (Subsegment) [inline tskv/src/reader/page.rs:33]: rm this

File: tskv/src/record_file/mod.rs [hunk 14]
```diff
@@ -68,7 +68,7 @@ pub const RECORD_DATA_TYPE_LEN: usize = 1;
 pub const RECORD_DATA_SIZE_LEN: usize = 4;
 pub const RECORD_CRC32_NUMBER_LEN: usize = 4;
 pub const RECORD_HEADER_LEN: usize = 14; // 4 + 1 + 1 + 4 + 4
-pub const BLOCK_SIZE: usize = 4096;
+pub const _BLOCK_SIZE: usize = 4096;
```
Comment (Subsegment) [inline tskv/src/record_file/mod.rs:71]: rm

File: tskv/src/schema/schemas.rs [hunk 15]
```diff
@@ -12,7 +12,7 @@ use models::schema::{
 use crate::database::FbSchema;
 use crate::schema::error::{Result, SchemaError};
 
-const TIME_STAMP_NAME: &str = "time";
+const _TIME_STAMP_NAME: &str = "time";
```
Comment (Subsegment) [inline tskv/src/schema/schemas.rs:15]: rm

File: tskv/src/summary.rs [hunk 16]
```diff
@@ -27,7 +26,7 @@ use crate::tsm::reader::TsmReader;
 use crate::version_set::VersionSet;
 use crate::{byte_utils, file_utils, ColumnFileId, LevelId, TseriesFamilyId};
 
-const MAX_BATCH_SIZE: usize = 64;
+const _MAX_BATCH_SIZE: usize = 64;
```
Comment (Subsegment) [inline tskv/src/summary.rs:29]: rm

File: tskv/src/tsm/mod.rs [hunk 17]
```diff
@@ -19,12 +19,12 @@ pub use tombstone::{Tombstone, TsmTombstone, TOMBSTONE_FILE_SUFFIX};
...
+pub(crate) const _MAX_BLOCK_VALUES: u32 = 1000;
 
-const HEADER_SIZE: usize = 5;
-const INDEX_META_SIZE: usize = 11;
-const BLOCK_META_SIZE: usize = 44;
-const BLOOM_FILTER_SIZE: usize = 64;
+const _HEADER_SIZE: usize = 5;
+const _INDEX_META_SIZE: usize = 11;
+const _BLOCK_META_SIZE: usize = 44;
+const _BLOOM_FILTER_SIZE: usize = 64;
```
Comment (Subsegment) [inline tskv/src/tsm/mod.rs:27]: rm

File: tskv/src/tsm/reader.rs [hunk 18]
```diff
@@ -79,7 +79,7 @@ impl TsmMetaData {
 
 #[derive(Clone)]
 pub struct TsmReader {
-    file_location: PathBuf,
+    _file_location: PathBuf,
```
Comment (Subsegment) [inline tskv/src/tsm/reader.rs:82]: rm

File: tskv/src/tsm/tombstone.rs [hunk 19]
```diff
@@ -25,8 +25,8 @@ use crate::record_file::{self, RecordDataType, RecordDataVersion};
 use crate::{byte_utils, file_utils, TskvError, TskvResult};
 
 pub const TOMBSTONE_FILE_SUFFIX: &str = "tombstone";
-const FOOTER_MAGIC_NUMBER: u32 = u32::from_be_bytes([b'r', b'o', b'm', b'b']);
-const FOOTER_MAGIC_NUMBER_LEN: usize = 4;
+const _FOOTER_MAGIC_NUMBER: u32 = u32::from_be_bytes([b'r', b'o', b'm', b'b']);
+const _FOOTER_MAGIC_NUMBER_LEN: usize = 4;
```
Comment (Subsegment) [inline tskv/src/tsm/tombstone.rs:29]: rm

File: tskv/src/wal/mod.rs [hunk 20]
```diff
@@ -57,27 +57,27 @@ use crate::tsm::codec::{get_str_codec, StringCodec};
...
+const _WAL_FOOTER_MAGIC_NUMBER_LEN: usize = 4;
 
-const SEGMENT_MAGIC: [u8; 4] = [0x57, 0x47, 0x4c, 0x00];
+const _SEGMENT_MAGIC: [u8; 4] = [0x57, 0x47, 0x4c, 0x00];
 
 /// A channel sender that send write WAL result: `(seq_no: u64, written_size: usize)`
-type WriteResultSender = oneshot::Sender<crate::TskvResult<(u64, usize)>>;
-type WriteResultReceiver = oneshot::Receiver<crate::TskvResult<(u64, usize)>>;
+type _WriteResultSender = oneshot::Sender<crate::TskvResult<(u64, usize)>>;
+type _WriteResultReceiver = oneshot::Receiver<crate::TskvResult<(u64, usize)>>;
```
Comment (Subsegment) [inline tskv/src/wal/mod.rs:80]: rm

File: tskv/src/wal/mod.rs [hunk 21]
```diff
@@ -393,16 +393,9 @@ mod test {
...
-        let mut fbb = flatbuffers::FlatBufferBuilder::new();
-        let ptr = models_helper::create_random_points_with_delta(&mut fbb, 5);
-        fbb.finish(ptr, None);
-        fbb.finished_data().to_vec()
-    }
-
     /// Generate flatbuffers data and memcache data
     #[allow(clippy::type_complexity)]
-    pub fn const_write_data(
+    pub fn _const_write_data(
```
Comment (Subsegment) [inline tskv/src/wal/mod.rs:398]: rm

File: tskv/src/compaction/compact.rs (hunk 2 above)
Comment (liikeuforever) [reply tskv/src/compaction/compact.rs]: thanks and done.

File: tskv/src/compaction/picker.rs (hunk 7 above)
Comment (Subsegment) [reply tskv/src/compaction/picker.rs:21]: rm this

File: tskv/src/reader/column_group/mod.rs (hunk 11 above)
Comment (liikeuforever) [reply tskv/src/reader/column_group/mod.rs:78]: This function is used in the test module.

File: tskv/src/reader/column_group/mod.rs [hunk 22]
```diff
@@ -330,11 +312,12 @@ mod tests {
             Field::new("c4", DataType::Boolean, true),
         ]));
 
-        let column_group_reader = ColumnGroupReader::new_with_unchecked(
```
Comment (Subsegment) [inline tskv/src/reader/column_group/mod.rs:333]: why clippy warning this?

File: tskv/src/compaction/picker.rs [hunk 23]
```diff
@@ -156,11 +151,8 @@ impl Picker for LevelCompactionPicker {
 }
 
 impl LevelCompactionPicker {
-    pub fn new(storage_opt: Arc<StorageOptions>) -> LevelCompactionPicker {
-        Self {
-            picking: AtomicBool::new(false),
-            storage: storage_opt,
-        }
+    pub fn new() -> LevelCompactionPicker {
+        Self {}
```
Comment (Subsegment) [inline tskv/src/compaction/picker.rs:155]: rm this

File: tskv/src/reader/column_group/mod.rs (hunk 22 above)
Comment (liikeuforever) [reply tskv/src/reader/column_group/mod.rs:333]: This function is only used once in the test module.
Modify to the following format to reduce the amount of code：
```
        let column_group_reader = ColumnGroupReader {
            column_group: Arc::new(ColumnGroup::new(0)),
            page_readers,
            schema,
            metrics: Arc::new(ExecutionPlanMetricsSet::new()),
        };
```

File: tskv/src/schema/schemas.rs [hunk 24]
```diff
@@ -257,34 +255,6 @@ impl DBschemas {
         }
     }
 
-    async fn check_update_table_res(
```
Comment (Subsegment) [inline tskv/src/schema/schemas.rs:260]: can we also rm check_create_table_res?

File: tskv/src/tsm/data_block.rs [hunk 25]
```diff
@@ -25,7 +25,7 @@ enum Merge {
 }
 
 impl DataBlock {
-    const BLOCK_SIZE: usize = 1000;
```
Comment (Subsegment) [inline tskv/src/tsm/data_block.rs:28]: rm this

File: tskv/src/tsm/mod.rs [hunk 26]
```diff
@@ -19,12 +19,8 @@ pub use tombstone::{Tombstone, TsmTombstone, TOMBSTONE_FILE_SUFFIX};
 
 use crate::tsm::data_block::DataBlock;
 
-pub(crate) const MAX_BLOCK_VALUES: u32 = 1000;
+pub(crate) const _MAX_BLOCK_VALUES: u32 = 1000;
```
Comment (Subsegment) [inline tskv/src/tsm/mod.rs:22]: rm this

File: tskv/src/wal/mod.rs [hunk 27]
```diff
@@ -57,27 +57,14 @@ use crate::tsm::codec::{get_str_codec, StringCodec};
...
 const WAL_FOOTER_MAGIC_NUMBER: u32 = u32::from_be_bytes([b'w', b'a', b'l', b'o']);
-const WAL_FOOTER_MAGIC_NUMBER_LEN: usize = 4;
-
-const SEGMENT_MAGIC: [u8; 4] = [0x57, 0x47, 0x4c, 0x00];
 
 /// A channel sender that send write WAL result: `(seq_no: u64, written_size: usize)`
-type WriteResultSender = oneshot::Sender<crate::TskvResult<(u64, usize)>>;
-type WriteResultReceiver = oneshot::Receiver<crate::TskvResult<(u64, usize)>>;
+type _WriteResultSender = oneshot::Sender<crate::TskvResult<(u64, usize)>>;
+type _WriteResultReceiver = oneshot::Receiver<crate::TskvResult<(u64, usize)>>;
```
Comment (Subsegment) [inline tskv/src/wal/mod.rs:67]: rm this

**Review by Subsegment — APPROVED** [review]

**Pull request merged**